compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4841,8 +4938,272 @@ public void visit(BLangTypedescExpr accessExpr, AnalyzerData data) {
         data.resultType = types.checkType(accessExpr, actualType, data.expType);
     }
 
+    public LinkedHashSet<BType> getBasicNumericTypesInFiniteType(BType referredType) {
+        BFiniteType finiteType = (BFiniteType) referredType;
+        Set<BLangExpression> valueSpace = finiteType.getValueSpace();
+        LinkedHashSet<BType> basicNumericTypes = new LinkedHashSet<>(valueSpace.size());
+
+        for (BLangExpression value : valueSpace) {
+            BType referredTypeInFiniteType = types.getReferredType(value.getBType());
+            int typeTag = referredTypeInFiniteType.tag;
+            if (TypeTags.isIntegerTypeTag(typeTag) || typeTag == TypeTags.FLOAT || typeTag == TypeTags.DECIMAL) {
+                if (TypeTags.isIntegerTypeTag(typeTag)) {
```

Comment:
`TypeTags.isIntegerTypeTag(typeTag)` is repeated. Why isn't this an if and else if instead?